cases/drp-00-05.tex

```diff
@@ -26,32 +34,149 @@ \subsubsection{Intercase dependencies}
+  \item{Given the LSST Science Pipelines version, the operator will generate the full config files and schema files (\S\ref{sec:configuration}), which are then ingested into the Data Backbone.}
+  \item{The pipeline workflow definition file of a consistent version will be downloaded; its operations- and dataset-specific inputs will be modified accordingly.}
+  \begin{enumerate}
+    \item{For LDM-503-2, the pipeline workflow definition file is a "wcl" file as used by the DESDM Framework Software.}
+  \end{enumerate}
+\end{enumerate}
+
+\paragraph{Execution}
+\begin{enumerate}
+  \item{The execution for each tract of the input data in \S\ref{sec:input} will be submitted to the hardware in \S\ref{sec:hardware} using the configuration in \S\ref{sec:configuration}.}
+  \item{During execution, the operator will use software to demonstrate the ability to check the processing status.}
+  \begin{enumerate}
+    \item{For LDM-503-2, the available Batch Production Service monitoring software consists of two commands: one to summarize currently submitted processing, one to get more details of single submission.}
+  \end{enumerate}
+  \item{If processing attempt completes, operator tags attempt as potential for release.  The campaign tags are used to make pre-release QA queries simpler.}
+  \item{If processing attempt fails, restart/resubmit on certain errors} \\
+    For LDM-503-2 test,
+    \begin{enumerate}
+        \item{Low level errors are automatically retried by framework (i.e., db connection fails, file transfer to/from data backbone fails).   For other errors, operator decides whether to just resubmit.   While a course-grained pipeline manual restart capability is available, the operator will simply resubmit the entire pipeline appropriately updating the state in the Oracle database.  Output files from previous attempts can be purged from the Data Backbone if no longer needed for debugging.}
+        \item{If a processing attempt is determined by the operator and production scientist to not be able to complete successfully, the attempt is not tagged and the problems are recorded.  All output files from a single failed processing attempt is kept in the Data Backbone for debugging purposes.} 
+    \end{enumerate}
+\end{enumerate}
+
+\paragraph{Non-science QA}
+\begin{enumerate}
+  \item{When the execution finishes, the success of the execution will be verified by checking the existence of the expected output data.
+  For each of the expected data products types (listed in \S\ref{drp-00-10-items})
+  and each of the expected units (PVIs, coadds, etc), the data product will be
+  verified to be non-empty.}
+    \begin{enumerate}
+        \item{Ask the Data Backbone about the files generated by each tract's execution through the pipeline.}
+        \item{Verify the physical and location information in Data Backbone DB matches the Data Backbone filesystem and vice-versa.} 
+    \end{enumerate}
```

Comment:
Do we do any check on the correctness of the Data Backbone, as opposed to just is consistency?

That is, my reading of the above is that it asks the Data Backbone what files were generated, and then it checks those files were consistently recorded on the filesystem and in the database. However, we should probably also check that list of files the Data Backbone things were generated is correct.

(If the DBB reports that no files were generated, and you verify that no files are recorded on the filesystem or in the database, you would seem to pass the above test without actually doing anything very useful!)

Reply:
The first part would verify that the "correct" files were generated by asking the DB, but it assumes the DB matches the filesystem.    The second part verifies that the filesystem matches the DB.